optuna_integration/wandb/wandb.py

```diff
@@ -136,6 +136,13 @@ def __init__(
             self._initialize_run()
 
     def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -> None:
+        # Handle case when trial has been pruned.
+        if trial.values is None:
+            if isinstance(self._metric_name, str):
+                trial.values = [None]
+            else:
+                trial.values = [None] * len(self._metric_name)
```

Comment:
I think we should not overwrite `trial.values` here. How about defining `values` and use it later logic?

```suggestion
        if trial.values is None:
            values = [None] * len(study.directions)
        else:
            values = trial.values
```

Note we cannot use `len(self._metric_name)` because this value can be different from the size of `values` as handled by https://github.com/optuna/optuna-integration/blob/15e6b0ec6d9a0d7f572ad387be8478c56257bef7/optuna_integration/wandb/wandb.py#L148-L155.

Reply:
> Note we cannot use len(self._metric_name) because this value can be different from the size of values as handled by

Hm.. this might be a little ugly, but my thinking was that when `trial.values` is `None` we really have no idea how many metrics were reported. So if the code automatically sets it to `len(self._metric_name_)` then for the run where no metrics were `report`-ed before it crashed, it really doesn't matter. 

The user would face the error for a run that actually works but reports an incorrect number of `trial.values`.
Is there any other cleaner way to access this information?

Reply:
In my understanding, by using `len(self._metric_name_)`, we cannot know the error with `values=None` trial with mismatch between `len(trial.values)` and `len(self._metric_name)`, but it will fail later trial with complete trial or non-None value, so I'm not sure why we use `len(self._metric_name_)`; it delayed the time when a user finds the bug.

> Is there any other cleaner way to access this information?
No, there is not.

Reply:
What do you think is the best solution here? if there's no way to access this information then really its just a tradeoff between having the user detect their error later vs. a single run crashing enough to bring down the `optuna` study.

That's a decision I guess only the maintainers can take ¯\_(ツ)_/¯

Reply:
Thank you for your comment. Hmm, to have a consistency with similar callback, https://github.com/optuna/optuna-integration/blob/main/optuna_integration/mlflow/mlflow.py, not logging `values` in the wandb run when `values` is `None`. So I'd like to suggest using the same approach as mlflow integration, which sounds similar to the first approach.

Reply:
Something like [this?](https://github.com/optuna/optuna-integration/blob/15e6b0ec6d9a0d7f572ad387be8478c56257bef7/optuna_integration/mlflow/mlflow.py#L236) I'm not sure if this is the best idea - if a run crashed out, wouldn't the user expect _something_ to be logged so that they can check? If the callbacks silently fails, I feel this could cause a lot of unnecessary trouble and pain for someone troubleshooting. Perhaps it might be better to log a `NaN` to wandb so that atleast there's a record of it?

Reply:
> Something like [this?](https://github.com/optuna/optuna-integration/blob/15e6b0ec6d9a0d7f572ad387be8478c56257bef7/optuna_integration/mlflow/mlflow.py#L236)
Yes.

At least params and other meta-data are recorded by wandb even though we don't call `wandb.log` for `values` as is mlflow integration. I don't think the suggested implementation silently fails. Could you explain when this happens?

Reply:
Well, I assume that if a run crashes the `values` is `None` then this will just.. not log anything to wandb. That's what I mean by the failure being silent. Personally, I'd rather see a red "FAILED" run in the wandb log then it silently just not doing anything. 

Perhaps I should add a warning? if you think its better to keep the error handling same across integrations that I will do that then.

Reply:
Sorry, I cannot understand the situation. Could you give some example?

Reply:
So if a run crashes before any metric is `report`-ed, the wandb callbacks will just do nothing as the `values = None`:
```py
if values is None:
    return None
```
Which I think could be a problem for the user as they'd have no idea this run crashed and how to fix the error.

Reply:
No, I meant not storing the `trial.values` to `metrics`, so the callback does store parameters, trial.number, trial.status, etc.